tests/gradients/test_finite_difference.py

```diff
@@ -0,0 +1,608 @@
+
+        x, y = 1.0 * params
+
+        expected = np.array([-np.sin(x) * np.sin(y), np.cos(x) * np.cos(y)])
+        assert np.allclose(jac, expected, atol=tol, rtol=0)
+
+        res = t.jacobian(jac, params)
+        expected = np.array(
+            [
+                [-np.cos(x) * np.sin(y), -np.cos(y) * np.sin(x)],
+                [-np.cos(y) * np.sin(x), -np.cos(x) * np.sin(y)],
+            ]
+        )
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    def test_tf_ragged(self, order, form, tol):
+        """Tests that the output of the finite-difference transform
+        of a ragged tape can be differentiated using TF, yielding second derivatives."""
+        tf = pytest.importorskip("tensorflow")
+        dev = qml.device("default.qubit.tf", wires=2)
+        params = tf.Variable([0.543, -0.654], dtype=tf.float64)
+
+        with tf.GradientTape() as t:
+            with qml.tape.JacobianTape() as tape:
+                qml.RX(params[0], wires=[0])
+                qml.RY(params[1], wires=[1])
+                qml.CNOT(wires=[0, 1])
+                qml.expval(qml.PauliZ(0))
+                qml.probs(wires=[1])
+
+            tape.trainable_params = {0, 1}
+            tapes, fn = finite_diff(tape, n=1, order=order, form=form)
+            jac = fn(dev.batch_execute(tapes))[1, 0]
+
+        x, y = 1.0 * params
+        res = t.gradient(jac, params)
+        expected = np.array([-np.cos(x) * np.cos(y) / 2, np.sin(x) * np.sin(y) / 2])
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    def test_torch(self, order, form, tol):
+        """Tests that the output of the finite-difference transform
+        can be differentiated using Torch, yielding second derivatives."""
+        torch = pytest.importorskip("torch")
+        from pennylane.interfaces.torch import TorchInterface
+
+        dev = qml.device("default.qubit", wires=2)
+        params = torch.tensor([0.543, -0.654], dtype=torch.float64, requires_grad=True)
+
+        with TorchInterface.apply(qml.tape.QubitParamShiftTape()) as tape:
+            qml.RX(params[0], wires=[0])
+            qml.RY(params[1], wires=[1])
+            qml.CNOT(wires=[0, 1])
+            qml.expval(qml.PauliZ(0) @ qml.PauliX(1))
+
+        tapes, fn = finite_diff(tape, n=1, order=order, form=form)
+        jac = fn([t.execute(dev) for t in tapes])
+        cost = torch.sum(jac)
+        cost.backward()
+        hess = params.grad
+
+        x, y = params.detach().numpy()
+
+        expected = np.array([-np.sin(x) * np.sin(y), np.cos(x) * np.cos(y)])
+        assert np.allclose(jac.detach().numpy(), expected, atol=tol, rtol=0)
+
+        expected = np.array(
+            [
+                [-np.cos(x) * np.sin(y), -np.cos(y) * np.sin(x)],
+                [-np.cos(y) * np.sin(x), -np.cos(x) * np.sin(y)],
+            ]
+        )
+        assert np.allclose(hess.detach().numpy(), np.sum(expected, axis=0), atol=tol, rtol=0)
+

```

Comment:
No ragged test for torch/jax?

Reply:
I think I realized (while writing the tests) that the ragged tests were a bit redundant; the underlying code that is being tested is the same regardless of autodiff library. So I was basically just re-testing existing tests in the `test/interfaces` folder